src/Frank/Builder.fs

```diff
@@ -10,43 +10,137 @@ module Builder =
     open Microsoft.AspNetCore.Http
     open Microsoft.AspNetCore.Routing
     open Microsoft.AspNetCore.Routing.Constraints
+    open Microsoft.AspNetCore.Routing.Template
     open Microsoft.Extensions.DependencyInjection
+    open Microsoft.OpenApi.Models
 
     let private methodNotAllowedHandler =
         RequestDelegate(fun ctx ->
             ctx.Response.StatusCode <- 405
             upcast Task.FromResult())
         |> RouteHandler
-
+    
+    /// Defines an HTTP resource with optional Open API documentation.
+    type Resource =
+        { RouteTemplate : string
+          Endpoints : Route[]
+          Description : OpenApiPathItem }
+    
+    /// Defines the Open API operation response documentation.
+    type OperationResponseMetadata =
```

Comment:
one thing I'd mention is that openapi allows for a combination of `(statusCode, contentType)` to pinpoint a response type, so that might need to be taken into account in this model.

Reply:
I'm thinking of stripping this piece out entirely. I'm not sure I know the best way to model apart from the actual `OpenApiOperation` type. I can at least start there and work out a better API later. Thinking of perhaps using another CE, actually.

Reply:
For the interim, I'm planning on writing some helpers to create an `OpenApiSchema` given a `Type`. For the rest, the existing OpenApi types seem decent for describing the operation.